lib/jormungandr/src/Cardano/Wallet/Jormungandr/Api/Types.hs

```diff
@@ -0,0 +1,183 @@
+
+data StakeApiResponse = StakeApiResponse
+    { epoch :: ApiT EpochNo
+    , stake :: ApiStakeDistribution
+    } deriving (Show, Eq, Generic)
+
+data ApiAccountState = ApiAccountState
+    { currentBalance :: !(ApiT (Quantity "lovelace" Word64))
+    , delegationStakePoolId :: !(Maybe (ApiT PoolId))
+    , totalTransactionCount :: !(ApiT (Quantity "transactionCount" Word64))
+    } deriving (Eq, Show, Generic)
+
+data ApiStakeDistribution = ApiStakeDistribution
+    { dangling :: ApiT (Quantity "lovelace" Word64)
+    , pools :: [(ApiT PoolId, ApiT (Quantity "lovelace" Word64))]
+    , unassigned :: ApiT (Quantity "lovelace" Word64)
+    } deriving (Eq, Show, Generic)
+
+instance ToHttpApiData BlockId where
+    toUrlPiece (BlockId (Hash bytes)) = decodeUtf8 $ convertToBase Base16 bytes
+
+instance ToHttpApiData (ApiT ChimericAccount) where
+    toUrlPiece (ApiT (ChimericAccount bytes)) =
+        decodeUtf8 $ convertToBase Base16 bytes
+
+instance MimeUnrender JormungandrBinary [BlockId] where
+    mimeUnrender _ =
+        pure . fmap (BlockId . Hash) . runGet (many $ getByteString 32)
+
+instance MimeUnrender Hex BlockId where
+    mimeUnrender _ bs =
+        BlockId . Hash <$> convertFromBase Base16 (BL.toStrict bs)
+
+-- | Polymorphic wrapper type to put around primitive types and, 3rd party lib
+-- types to avoid defining orphan instances and/or, undesirable instances on
+-- primitive types. It helps to keep a nice separation of concerns between the
+-- API layer and other modules.
+newtype ApiT a =
+    ApiT { getApiT :: a }
+    deriving (Generic, Show, Eq)
+
+{-------------------------------------------------------------------------------
+                                Content Types
+-------------------------------------------------------------------------------}
+
+-- | Represents the binary format of Jörmungandr.
+data JormungandrBinary
+
+instance Accept JormungandrBinary where
+    contentType _ = contentType $ Proxy @Servant.OctetStream
+
+instance MimeUnrender JormungandrBinary Block where
+    mimeUnrender _ = pure . runGet getBlock
+
+instance MimeRender JormungandrBinary (Tx, [TxWitness]) where
+    mimeRender _ (Tx _ ins outs, wits) =
+        runPut $ withHeader MsgTypeTransaction $ putSignedTx ins outs wits
+
+data Hex
+
+-- | Represents data rendered to hexadecimal text.
+instance Accept Hex where
+    contentType _ = contentType $ Proxy @Servant.PlainText
+
+{-------------------------------------------------------------------------------
+                              FromJson instances
+-------------------------------------------------------------------------------}
+
+instance FromJSON StakeApiResponse where
+    parseJSON = genericParseJSON defaultOptions
+
+instance FromJSON ApiAccountState where
+    parseJSON = withObject "ApiAccountState" $ \v ->
+        ApiAccountState
+            <$> v .: "value"
+            <*> v .:? "delegation"
+            <*> v .: "counter"
```

Comment:
This should be updated to decode this format:
```json
{ "counter": 0
, "delegation":
  { "pools":
    [ [ "5cfbb7f7ec0050499fbd1c79515216c5ad636b97d1944bde8a7db98e7d126803"
      , 1
      ]
    ]
  },
  "value": 1
}
```
See: https://github.com/input-output-hk/cardano-wallet/issues/902#issuecomment-552382037